src/server/main_service.cc

```diff
@@ -1167,12 +1167,19 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
 
 void Service::Publish(CmdArgList args, ConnectionContext* cntx) {
   string_view channel = ArgS(args, 1);
-  string_view message = ArgS(args, 2);
+
+  // To ensure that this message that we are process in one context will live long enough
+  // when passing over to other contexts we need to have shared ownership.
+  // This use to have a manual ref counter. But it is replaced here with shared pointer that
+  // getting the same result.
+  // Please note that this value (the message) will span across multiple connections which live
+  // across many threads (see the comment bellow regarding connection handling).
+  std::shared_ptr<std::string_view> message = std::make_shared<std::string_view>(ArgS(args, 2));
```

Comment:
Yes, there is a problem with using the string_view.
The problematic scenario:
1. You call publish_cb that adds the message (string_view) to the dispatch queue
2. The `Await` call finishes and exits the scope of this function. the args object and its backing memory disappear.
3. another thread pulls the string_view object from the dispatch queue and accesses it.

Reply:
I've changed this to use string and now string_view so now the memory is owned by the shared_ptr (as the string inside it allocate the memory).
Do you think that we need some custom allocator for this as well?

Reply:
no, I think it's ok now.

Reply:
ok